app/services/database_operations/database_operations.py

```diff
@@ -131,25 +135,62 @@ def insert_into_robots_db(
     session.add(robot_profile)
     session.commit()
 
-def update_user_location(longitude, latitude, user):
-    session = Session()
 
-    new_user_location = WKTElement(f'POINT({longitude} {latitude})', srid=4326)
+def update_user_location(longitude, latitude, user):
+    new_user_location = WKTElement(f"POINT({longitude} {latitude})", srid=4326)
 
-    session.query(User).filter(User.username == user).update({User.location: new_user_location}, synchronize_session=False)
+    user.location = new_user_location
 
     session.commit()
 
 
 def select_all_from_database(model_class):
-    session = Session()
-    all_information_from_db = session.query(model_class).all()
-    return all_information_from_db
+    return session.query(model_class).all()
+
+
+def get_user(user):
+    return session.query(User).filter_by(username=user).first()
+
+
+def get_not_matched_robots_by_localization(user):
+    distance = 50 * 1000  # 50 km przeliczone z metrów
+    MAX_ROBOTS = 50
+
+    robots_within_distance_not_matched = (
+        session.query(UserRobot)
+        .options(
+            joinedload(UserRobot.profile_robot)  # Dołączamy profile w jednym zapytaniu
+        )
+        .filter(
+            # Filtrowanie robotów, którzy nie są w tabeli Matches dla danego użytkownika
+            not_(
+                UserRobot.id.in_(
+                    session.query(Matches.robot_id).filter(Matches.user_id == user.id)
+                )
+            ),
+            # Filtrowanie robotów w określonej odległości od punktu odniesienia
+            ST_DWithin(UserRobot.location, user.location, distance),
+        )
+        .limit(MAX_ROBOTS)
+        .all()
+    )  # Ustawienie limitu robotów
+
+    return robots_within_distance_not_matched
+
+
+def add_match(user_id, robot_id):
+    new_match = Matches(user_id=user_id, robot_id=robot_id)
+    session.add(new_match)
+    session.commit()
+    return
+
+
+def get_user_criteria(user):
+    user_criteria = (
+        session.query(UserCriteria).filter(UserCriteria.user_id == user.id).first()
+    )
+    return user_criteria
 
 
-def get_user_from_User_table(user):
-    session = Session()
-    selected_user = session.query(User).filter_by(username=user).first()
-    #domicile = session.query(Profile.domicile).join(User, User.id == Profile.user_id).filter(User.username == user).first()
-    #selected_user = [user_in_db, domicile]
-    return selected_user
+def download_matched_users(user, reference_point):
+    pass

```

Comment:
download is not a good name i iguess, how about get or retrieve

Reply:
Done